### DISCUSSION TITLE
How to choose the backend server(webapp) and set it to http header "X-Upstream-Server" and then send the request to a proxy?

### DISCUSSION THREAD

**Discussion by ralgond** [nginx/nginx#272] [Q&A] [answered]
Hi, there:

My service‘s architect is as following:

user<---->nginx<---->ratelimiter<---->webapps. 

Now I do not want to add a load balancer to the ratelimiter, so I wish to make nginx to choose a webapp and tell the ratelimiter to connect.

**Comment by HanadaLee** [comment, marked as answer]
You can considor ngx_http_split_clients_module. the following is a simple example, it will be hashed according to the specified percentage based on $remote_addr
```
http {
    split_clients $remote_addr $real_upstream_webapp {
        0.5%    a.webapp-server.com;
        2.0%    b.webapp-server.com;
        *       default.webapp-server.com;
    }
    upstream ratelimiter {
        server 127.0.0.1:8080;
    }

    server {
        location / {
            proxy_set_header X-Upstream-Server $real_upstream_webapp;
            proxy_pass http://ratelimiter;
        }
    }
}
```
Visit the documentation page for a detailed description of the module: https://nginx.org/en/docs/http/ngx_http_split_clients_module.html
